File: leetcode/lc-us/cpp/1572.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int diagonalSum(vector<vector<int>> &mat) {
        int n = (int) mat.size();
        int center = mat[n / 2][n / 2];
        int sum = 0;
        for (int i = 0; i < n; i++) {
            sum += mat[i][i] + mat[i][n - 1 - i];
        }
        if ((n & 1) == 1) {
            sum -= center;
        }
        return sum;
    }
};

File: leetcode/lc-us/rust/1572.rs
impl Solution {
    pub fn diagonal_sum(mat: Vec<Vec<i32>>) -> i32 {
        let mut sum = 0;
        for i in 0..mat.len() {
            sum += mat[i][i] + mat[i][mat.len() - 1 - i]
        }
        if mat.len() % 2 == 1 {
            sum -= mat[mat.len() / 2][mat.len() / 2]
        }
        sum
    }
}